### PR TITLE
feat(packages): ts-config | eslint | prettier  

### LINKED ISSUE #40: eslint + tslint + prettier shared configs
**Is your feature request related to a problem? Please describe.**
Currently we have tsconfigs and eslint configs in different locations.

**Describe the solution you'd like**
We should have shared config packages.

**Describe alternatives you've considered**
Leave as is.

### PR DESCRIPTION
**Pull request by devs9**
## Scope

[closes issue-#40](https://github.com/crherman7/rechunk/issues/40)

## Implementation

* created eslint shared configs
* created prettier shared configs
* created tsconfig shared configs

## Screenshots

<details>
  <summary>ios</summary>

  https://github.com/user-attachments/assets/7bf216e7-92c3-4818-b815-79687f2b2a8e
</details>

<details>
  <summary>android</summary>
  
  https://github.com/user-attachments/assets/8277f073-a306-4832-a15b-8275b4389a60
</details>

## How to Test

* go to any example and run `lint` script
* or run `pnpm eslint --print-config ${path to file}` to see `eslint` config

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "registry": "https://registry.npmjs.org"
   },
   "scripts": {
-    "build": "tsup",
+    "build": "cd packages/api-client && pnpm build && cd ../../ && tsup && pnpm --filter '@repo/*' recursive run --sort build",
```
Comment (devs9) [inline package.json:37]: 🟡 @crherman7 I added this as a temporary solution before merging your work with the migration. but let me know if it's not necessary

File: packages/eslint-config/package.json [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
-  "private": true,
   "author": "Chris Herman",
   "keywords": [],
-  "main": "index.js",
+  "private": true,
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build": "pnpm tsc"
+  },
+  "exports": {
```
Comment (crherman7) [inline packages/eslint-config/package.json:12]: package exports very nice 💚

File: package.json (hunk 1 above)
Comment (crherman7) [reply package.json:37]: yeah I will need to test this out

Comment (crherman7) [reply package.json:37]: yeah this workaround does make sense for the temporary workflow

File: apps/react-native/package.json [hunk 3]
```diff
@@ -31,8 +35,6 @@
     "@react-native/typescript-config": "0.73.1",
```
Comment (crherman7) [inline apps/react-native/package.json]: 🟡 Is `@react-native/typescript-config` and `@react-native/eslint-config` still needed?

Comment (devs9) [reply apps/react-native/package.json]: good catch 👍

**Review by crherman7 — APPROVED** [review]
Awesome job 💚🔥💯 this will be massively useful!

**Pull request merged**